penguin/core.py

```
@@ -2149,6 +2155,32 @@ def _get_session_token_usage(
 
         return usage
 
+    @staticmethod
+    def _preserve_session_usage_truncations(
+        usage: Dict[str, Any], usage_snapshot: Any
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `usage_snapshot`

(ANN401)